src/test/java/pro/trevor/tankgame/e2e/TestEndToEndV3.java

```diff
@@ -0,0 +1,70 @@
+package pro.trevor.tankgame.e2e;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import pro.trevor.tankgame.rule.impl.ruleset.DefaultV3RulesetRegister;
+import pro.trevor.tankgame.state.attribute.Attribute;
+import pro.trevor.tankgame.state.board.Position;
+
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static pro.trevor.tankgame.e2e.EndToEndTestUtils.*;
+
+public class TestEndToEndV3 {
+
+    private EndToEndTester tester;
+
+    @BeforeEach
+    public void initialize() {
+        tester = new EndToEndTester(new DefaultV3RulesetRegister(), "src/test/resources/initial-v3.json", "src/test/resources/moves-v3.json");
+    }
+
+    @Test
+    public void testStateAtEndState() {
+        assertFalse(Attribute.RUNNING.unsafeFrom(tester.getState()));
+        assertEquals("Corey", Attribute.WINNER.unsafeFrom(tester.getState()));
+        assertEquals(18, Attribute.TICK.unsafeFrom(tester.getState()));
+        assertEquals(15, tester.getCouncil().allPlayersOnCouncil().size());
+        assertEquals(20, Attribute.COFFER.unsafeFrom(tester.getCouncil()));
+
+        tester.getCouncil().getCouncillors().forEach((p) -> assertFalse(tester.getCouncil().isPlayerSenator(p)));
```

Comment:
These seems like a generic checks that would happen on most versions.  Helper function?